pants-plugins/examples/bash/create_binary.py

```diff
@@ -0,0 +1,78 @@
+# Copyright 2020 Pants project contributors.
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from dataclasses import dataclass
+
+from pants.core.goals.binary import BinaryFieldSet, CreatedBinary
+from pants.core.goals.run import RunRequest
+from pants.core.util_rules.source_files import SourceFiles, SourceFilesRequest
+from pants.engine.addresses import Addresses
+from pants.engine.process import BinaryPathRequest, BinaryPaths, Process, ProcessResult
+from pants.engine.rules import Get, collect_rules, rule
+from pants.engine.target import Dependencies, TransitiveTargets
+from pants.engine.unions import UnionRule
+from pants.util.logging import LogLevel
+
+from examples.bash.target_types import BashBinarySources, BashSources
+
+
+@dataclass(frozen=True)
+class BashBinaryFieldSet(BinaryFieldSet):
+    required_fields = (BashBinarySources,)
+
+    sources: BashBinarySources
+    dependencies: Dependencies
+
+
+@rule(level=LogLevel.DEBUG)
```

Comment:
I think that this is the default level.

Reply:
The default is now Trace. See https://github.com/pantsbuild/pants/pull/10639/files#r471816205.